python/lsst/ts/externalscripts/auxtel/latiss_acquire.py

```diff
@@ -0,0 +1,604 @@
+    def get_best_effort_isr(self):
+        # Isolate the BestEffortIsr class so it can be mocked
+        # in unit tests
+        return BestEffortIsr()
+
+    # TODO: move to new class as part of DM-37665
+    async def get_next_image_data_id(self, flush=True):
+        """Return dataID of image that appears from the ATOODS CSC.
+
+        This is meant to be called at the same time as a take image command.
+        If this is called after take_image is completed, it may not receive
+        the imageInOODS event.
+
+        Parameters
+        ----------
+        timeout: `float`
+            Amount of time to wait for image to arrive.
+        """
+
+        self.log.debug(
+            f"Waiting for image to arrive in OODS for a maximum of {self.image_in_oods_timeout} seconds."
+        )
+        in_oods = await self.latiss.rem.atoods.evt_imageInOODS.next(
+            timeout=self.image_in_oods_timeout, flush=flush
+        )
+
+        data_id = parse_obs_id(in_oods.obsid)
+        self.log.info(f"seq_num {data_id['seq_num']} arrived in OODS")
+
+        return data_id
+
+    async def find_offset(self, data_id, target_position):
+        """Detects the brightest star in an image and returns the offsets
+        (dx_arcsec and dy_arcsec) required to move the target to the target
+        position.
+
+        Parameters
+        ----------
+        data_id: `dict`
+            data Id for initial exposure with target initial position.
+        target_position: `array`
+            desired final position in detector coordinates as array (X,Y).
+
+        Returns
+        -------
+        dx_arcsec, dy_arcsec: `float`, `float`
+            offsets (arcsec) required to translate target from detected
+            position to desired final position
+        """
+        exp = await get_image(
+            data_id,
+            self.best_effort_isr,
+            timeout=self.acq_exposure_time + STD_TIMEOUT,
+        )
+
+        # Find brightest star in image.
+        loop = asyncio.get_event_loop()
+        executor = concurrent.futures.ThreadPoolExecutor()
+        result = await loop.run_in_executor(executor, self.qm.run, exp)
+
+        # Verify a result was achieved, if not raise the exception.
+        if not result.success:
+            raise RuntimeError("Centroid finding algorithm was unsuccessful.")
+
+        current_position = PointD(
+            result.brightestObjCentroid[0], result.brightestObjCentroid[1]
+        )
+
+        # Find offsets to desired position
+        self.log.debug(
+            f"Current brightest target position is {current_position} whereas the "
+            f"target position is {self.target_position}"
+        )
+
+        dx_arcsec, dy_arcsec = calculate_xy_offsets(current_position, target_position)
+        return dx_arcsec, dy_arcsec
+
+    async def execute_acquisition(self, target_position):
+        """Perform acquisition of target. Will iteratively measure position of
```

Comment:
Make sure the first phrase stands alone in the first line, providing a summary of the method description, e.g.:

```
"""Perform acquisition of target. 

Will iteratively measure position of...
```